### PR TITLE
Feat map

### PR DESCRIPTION
**Pull request by mixcas**
Enqueues Google Maps API
Inits the map and add markers
Play location videos when a marker is clicked

### CODE REVIEW

File: lib/theme-options/site-options.php [hunk 1]
```diff
@@ -62,6 +62,12 @@
       'id'   => $prefix . 'og_fb_app_id',
       'type' => 'text',
     ),
+    array(
+      'name' => __( 'Google API Key', 'IGV' ),
```
Comment (varffs) [inline lib/theme-options/site-options.php]: This is not a Metadata Option. It should have its own section at the top of the setting page

File: src/js/main.js [hunk 2]
```diff
@@ -355,4 +357,87 @@ Site.Portraits = {
...
+
+  init: function() {
+    var _this = this;
+
+    // Get grid element
+    // TODO: find a way of not repeating this data here and in Site.Portraits
+    _this.$grid = $('#portraits-grid');
+
+    // Get portraits
+    // TODO: find a way of not repeating this data here and in Site.Portraits
```
Comment (varffs) [inline src/js/main.js:381]: just save it on `Site`

File: src/js/main.js [hunk 3]
```diff
@@ -355,4 +357,87 @@ Site.Portraits = {
...
+
+    // Get map container
+    _this.$container = $('#portraits-map');
+
+    // Init google maps
+    _this.map = new google.maps.Map(_this.$container[0], _this.options);
+
+    // Add a marker for each location
+    if (WP.locations.length) {
+      for(var i = 0; i < WP.locations.length; i++) {
```
Comment (varffs) [inline src/js/main.js]: you can use jQuery each here if you want (aka why not)

File: src/js/main.js [hunk 4]
```diff
@@ -355,4 +357,87 @@ Site.Portraits = {
...
+    // Save youtube IDs
+    $filteredPortraits.each(function(index) {
+      list[index] = this.dataset.youtubeId;
+    });
+
+    // Load list
+    Site.Player.setVideosList(list);
+
+    // Play first video
+    Site.Player.playVideo(list[0]);
```
Comment (varffs) [inline src/js/main.js]: I feel like these last 2 function calls to the Player should be one function on the player like `playList(list)` or something

File: src/js/main.js [hunk 5]
```diff
@@ -355,4 +357,87 @@ Site.Portraits = {
...
+
+    // TODO: Scroll to player
```
Comment (varffs) [inline src/js/main.js]: Fine to be splitting things up like this but we need a better project management solution not to lose them in the mix. Perhaps someone in this flow needs to make these TODOs into Trello cards? or GH issues?

**Review by varffs — CHANGES_REQUESTED** [review]
Not sure the overall UX for when you select a location. I feel like it should also filter down the grid view (as it will be possible to move between those views once states like location have been toggled yh?)

(The change is the metabox location)

File: src/js/main.js (hunk 5 above)
Comment (mikeyrayvon) [reply src/js/main.js]: the styling branch for this comes next

**Comment by mikeyrayvon** [comment]
@mixcas  and I discussed the interaction between Map and Grid filters yesterday. We decided to give them separate playlists, so the Map doesnt effect the Grid and visa versa.

Use case: you have a filtered list on the Grid and you're playing a video, then switch to the Map view before the video ends. Video closes, Map slides into view. From there you can play videos by location. If you switch back to the Grid view your filtered list is still displayed, but hasn't been effected by Map.

File: src/js/main.js (hunk 3 above)
Comment (mixcas) [reply src/js/main.js]: Word, i tried it on another array down below. Will change.

File: src/js/main.js (hunk 5 above)
Comment (mixcas) [reply src/js/main.js]: I'm trying to use code TODOs dor things that need to happen eventually but fall out of scope. I normally add TODOs as a way to remind whoever makes the next feature  or knows where to do it really fast.

In this case, scrollingTo comes in a diff branch cuz it depends on other parts that need to be made yet.

Comment (mixcas) [reply src/js/main.js]: I should have explained that first

**Comment by mixcas** [comment]
@varffs we agreed on this interaction to start with and see how it feels. I feel that from this base it could be changed to something like sharing the list or something else very easly. What do you think?

File: src/js/main.js (hunk 5 above)
Comment (varffs) [reply src/js/main.js]: All the same we've finished projects and shipped them with TODO comments left in the code. Also most of these todos are nice little bitsize features. So would be good to have a list of them somewhere for memory and also for quick work to be done

**Comment by varffs** [comment]
I would think to do the UX the other way round. But fine if we are doing it with way and then testing. I'll wait for those details and merge

File: src/js/main.js (hunk 5 above)
Comment (mixcas) [reply src/js/main.js]: Mmmmm maybe, i cant remember. Nothing recent, I think. I know what you are saying.

With ship projects with issues too sometimes. Either way its not the system, cuz we have a system for issues, it's really us. 

These TODO have worked for me when I do the feat that matches a TODO, that's how I've been using them.

What could be done is to add a note, bullet or something to the corresponding trello card for the feat task. In this case, add a bullet to the style task in the trello board.

Comment (mixcas) [reply src/js/main.js]: I'd still use the comment TODO cuz its useful when coding

Comment (varffs) [reply src/js/main.js]: Not suggesting to remove the code comments, just that perhaps we could track them or note them somewhere else so they dont get forgotten. Anyhow the scrollto function isn't styling but a clear feature after styling feature is done

Comment (mixcas) [reply src/js/main.js]: Word, adding to trello makes sense, no?

File: lib/functions-custom.php [hunk 6]
```diff
@@ -88,3 +88,32 @@ function get_post_filters_data($post) {
...
+  if (!empty($locations)) {
+
+    $data = array();
+
+    foreach($locations as $location) {
+      // Get coordinates
+      $lat = get_term_meta($location->term_id, '_igv_location_lat', true);
+      $lon = get_term_meta($location->term_id, '_igv_location_lon', true);
+
+      if (!empty($lat) && !empty($lat)) {
```
Comment (mixcas) [inline lib/functions-custom.php]: HA! nobody catched this one!

File: src/js/main.js (hunk 2 above)
Comment (mixcas) [reply src/js/main.js:381]: Yeah, i will move some stuff up to Site, but in a diff branch later

**Comment by mixcas** [comment]
@varffs ready

File: src/js/main.js [hunk 7]
```diff
@@ -387,7 +389,7 @@ Site.Map = {
 
     // Add a marker for each location
     if (WP.locations.length) {
-      for(var i = 0; i < WP.locations.length; i++) {
+      $(WP.locations).each(function(i) {
```
Comment (varffs) [inline src/js/main.js]: You will get `(index, item)` in this callback if you want. So you don't need the `WP.locations[i]` part. Also I think `i` is a valid name for an iterator but when using the callback like this `index` is nicer. But that is just style

File: src/js/main.js [hunk 8]
```diff
@@ -425,15 +427,12 @@ Site.Map = {
     var $filteredPortraits = _this.$portraits.filter(filter);
 
     // Save youtube IDs
-    $filteredPortraits.each(function(index) {
-      list[index] = this.dataset.youtubeId;
+    $filteredPortraits.each(function(i) {
+      list[i] = this.dataset.youtubeId;
```
Comment (varffs) [inline src/js/main.js]: lol

**Review by varffs — APPROVED** [review]

File: src/js/main.js (hunk 7 above)
Comment (mixcas) [reply src/js/main.js]: hehehe, k will use index and item

**Comment by mixcas** [comment]
done

**Pull request merged**